README.md
# Casino
A simulation of typical casino games. 


Change_Log.txt
##############################
Program Name: Casino
Program Description:To simulate simple casino games
Date Created: October 6, 2019
Date Completed: January 5, 2020
#############################
TODO - If I come back to this
Poker
UI
Actual program not just py files

#############################
Change Log
October 6, 2019 - Created and completed high_low() and craps() functions
October 7, 2019 - Created and completed recognize_number() function
October 9, 2019 - Fixed recognize_number() function to account for test case "10"
October 11, 2019 - Created and completed deck_of_cards() and draw_card() functions. Started a basic ui and black_jack() function. Organized functions
October 12, 2019 - Created and completed the skeleton of roulette function(). Worked a little on black_jack() and ui()
October 13, 2019 - Worked on roulette() function. Created and completed recognize_cardbj()
October 20, 2019 - Finished black_jack()
November 2, 2019 - Created a Login/Create Account system with comments
December 20, 2019 - Reworked games to return int values and finished roulette()
December 21, 2019 - Recreated Login/Logout system and create account so that it is not on the same screen as the games
December 24, 2019 - Created delete account, accounted for duplicate usernames when creating accounts, allowed to log into a different account while in ui. Reduced some clutters of code
December 27, 2019 - Edited login info to allow for new account creation and added parts of when one selects to continue
December 28, 2019 - Edited the user's bet for continue
January 1, 2020 - Worked on selctions and worked on roulette table in another program
January 3, 2020 - Worked on roulette decision
January 4, 2020 - Finished roulette decision
January 5, 2020 - Combines the roulette decision and roulette table into this program
##############################

main.py
import card_game
import other_games
import os
import roulette
from termcolor import *

def login_delete(lists, ld):
  user_user = input("Enter username: ")
  count = 1
  marker = 0 

  #Checks is username in database
  while len(lists) > count:
    if user_user == lists[count][:len(user_user)]:
      marker = count
      count = len(lists)    

    count += 1

  print("=====")

  if marker == 0:
    print("User not in database")
    return 0

  else:
    logs = lists[marker].split(",")

    user_pass = input("Enter password: ")

    while user_pass != logs[1]:
      print("Passwords do not match")
      print("=====")
      user_pass = input("Enter password or enter # to leave : ")

      if user_pass == "#":
        print("Leaving")
        return 0

    os.system('clear')

    if ld == "login":
      print("User has logged in")
    else:
      print("User is deleted")

    return marker

def account_creation(lists) -> None:
    #Username
    valid_user = False
    print("Your name can only include alphanumeric characters.")
    name = input("Enter your username: ")
    while not valid_user:
      #If not valid username
      if name.isalnum():
        valid_user = True
      
      #If not the only account
      if len(lists) > 1:
        in_database = True
        for i in range(1, len(lists)):          
          if lists[i][0:lists[i].find(",")] == name:
            in_database = False
            valid_user = False
      
      #Checkers
      if not in_database:
        print("=====")
        print("User is already in database.")
        name = input("Enter your username: ")

      elif not valid_user:
        print("=====")
        print("Enter a valid username.")
        name = input("Enter your username: ")      

    #Password
    print("=====")
    print("Your password can only include alphanumeric characters.")
    print("Your password must also have a mimnimum length of 8.")
    password = input("Enter your password: ")

    while len(password) < 8 or not password.isalnum():      
      print("=====")
      print("Password is invalid.")
      password = input("Enter your password: ")

    #Reenter password
    os.system('clear')  
    re_enter = input("Reenter your password or enter ""#"": ")

    while re_enter != password:
      if re_enter == "#":
        print("Account not created")
        print("=====")
        return

      os.system('clear')
      print("Passwords do not match")      
      re_enter = input("Reenter your password or enter ""#"": ")
      
    os.system('clear')
    
    lists[0] = "0\n"
    lists.append(name + "," + password + "," + "200\n")
    print("Account created. Account contains $200 on the house.")
    print("Login in to continue")
    print("=====")

def selector() -> str:
    print("Continue: 0")
    print("Log out: 1")
    print("Create a new account: 2")
    print("Log into a different account: 3")
    print("Delete account: 4")
    print("Exit: 5")

    return input("Select 1 of the above to continue: ")
    print("=====")

def greeter(user_info):
  print("Greetings " + user_info[0] + ".")
  print("You have $" + user_info[2].strip() + ".")
  print("=====")


def bet_maker(cash):
  bet = input("Enter your bet amount: ")

  if not bet.isnumeric() or int(bet) > cash or int(bet) <= 0:
    print("Enter a valid bet amount.")
    return -1

  return int(bet)

def bet_outcome(cash):
  if cash == 0:
    print("No changes were made to your balance")
  elif cash > 0:
    print("You have gained $" + str(cash) + ".")
  else:
    print("You have lost $" + str(-1 * cash) + ".")
  

def ui() -> None:
  #Files
  reader = open("data.txt", "r")
  info = reader.readlines()  
  reader.close()

  writer = open("data.txt", "w")

  #No accounts
  if len(info) == 1:
    print("There are no accounts open at this current moment.")
    print("Please create one to continue.")
    account_creation(info)

  iden = int(info[0].strip())

  #No logged in account
  while iden == 0:
    logged_in = False
    while not logged_in:
      print("No user has logged in.")
      print("Login in or create a new account to continue")
      print("Login: 1")
      print("Create account: 2")
      log_acc = input("Select 1 to continue: ")     

      while not(log_acc == "1" or log_acc == "2"):
        print("Enter a valid input")
        log_acc = input("Select 1 to continue: ")
      
      if log_acc == "2":
        account_creation(info)

      else:
        iden = login_delete(info, "login")

        if iden != 0:
          logged_in = True

      
    info[0] = str(iden) +"\n"
    print("=====")

  user_info = info[iden].split(",")
  greeter(user_info)

  #User select
  exited = True
  decision = selector()

  if decision == "5":
    print("Exiting of casino...")
    exited = False  

  while exited:
    #Continue
    if decision == "0":
      cash = int(user_info[2].strip())
      old = cash

      play = "0"

      while play != "5":
        print("=====")
        print("You have $" + str(cash) + ".")
        print("Backjack: 1")
        print("Craps: 2")
        print("Roulette: 3")
        print("Highlow: 4")
        print("Exit: 5")

        play = input("Select the game you want to play: ")
        print("=====")

        bet = 0

        if play == "1":
          bet = bet_maker(cash)

          if bet != -1:  
            bet = card_game.black_jack(bet)
            bet_outcome(bet)          
            cash += bet

            if cash <= 0:
              print("You do not have anymore money.")        
              print("Account will now be deleted.")
              print("=====")

              info[-1], info[iden] = info[iden], info[-1]
              info = info[:-1]

              play = "5"
              exited = False
              iden = 0
              info[0] = "0\n"

        elif play == "2":
          bet = bet_maker(cash)

          if bet != -1:  
            guess = input("Enter if shooter will win or lose. input 'Win' for shooter win and anything else for shooter lose: ")
            bet = other_games.craps(guess, bet)
            bet_outcome(bet)
            cash += bet

            if cash <= 0:
              print("You do not have anymore money.")        
              print("Account will now be deleted.")
              print("=====")

              info[-1], info[iden] = info[iden], info[-1]
              info = info[:-1]

              play = "5"
              exited = False
              iden = 0
              info[0] = "0\n"

        elif play == "3":
          bet = bet_maker(cash)

          if bet != -1:
            roulette.table()
            guess = roulette.choices()
            if guess[-1] == 1:
              bet = other_games.roulette(guess[0], bet, guess[1])
              bet_outcome(bet)
              cash += bet
              if cash <= 0:
                print("You do not have anymore money.")        
                print("Account will now be deleted.")
                print("=====")

                info[-1], info[iden] = info[iden], info[-1]
                info = info[:-1]

                play = "5"
                exited = False
                iden = 0
                info[0] = "0\n"
            

          print("=====")

        elif play == "4":
          bet = bet_maker(cash)
          if bet != -1:
            guess = input("Enter if the number will be 'high' or 'low' than 50: ")
            bet = other_games.high_low(guess, bet)
            bet_outcome(bet)
            cash += bet

            if cash <= 0:
              print("You do not have anymore money.")        
              print("Account will now be deleted.")
              print("=====")

              info[-1], info[iden] = info[iden], info[-1]
              info = info[:-1]

              play = "5"
              exited = False
              iden = 0
              info[0] = "0\n"

        elif play == "5":
          print("You have $" + str(cash) + ".")
         
        else:
          print("Enter a valid input")

      if old != cash and iden != 0:
        user_info[2] = str(cash)
        infos = ""

        for i in user_info:
          infos += i + ","

        info[iden] = infos[:-1] + "\n"

      if iden != 0:
        decision = selector()

    #Log out
    elif decision == "1":
      info[0] = "0\n"
      exited = False
      print("=====")
      print("Logging out...")
      print("Program will now close")

    #Create account
    elif decision == "2":
      account_creation(info)
      decision = selector()
    
    #Log into different account
    elif decision == "3":
      info[0] = str(login_delete(info, "login")) +"\n"
      iden = int(info[0].strip())
      user_info = info[iden].split(",")

      greeter(user_info)

      decision = selector()

    #Delete account
    elif decision == "4":
      print("Enter the valid information to delete the account")
      delete = login_delete(info, "delete")
      if delete == 0:
        os.system('clear')
        print("Account not deleted")
      else:
        info[delete], info[-1] = info[-1],info[delete]
        info = info[0:-1]

      #If no more acocunts
      if len(info) == 1:
        print("No more accounts left, exiting program.")
        info[0] = 0

      else:
        decision = selector()      

    #Leave
    elif decision == "5":
      print("Exiting of casino...")
      exited = False

    #Invalid selection
    else:
      print("Enter a valid input to continue")
      print("=====")
      decision = selector()


  #Writes to file all users
  for i in range(0, len(info)):
    writer.write(info[i])
  
  writer.close()

if __name__ == "__main__":
  ui()

  

cards.py
def deck_of_cards():
  """
  Returns a list that mimics a deck of cards  
  """
  suits = ["Diamonds", "Clubs", "Hearts", "Spades"]
  values = ["Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King",]
  deck = []

  for i in suits:
    for j in values:
      deck.append(j + " of " + i)

  return deck

def recognize_numberbj(card: str) -> int:
  """
  Returns a number based on card's value in blackjack
  >>> recognize_numberbj("Ace of hearts")
  11
  >>> recognize_numberbj("King of spades")
  10
  >>> recognize_numberbj("2 of clubs")
  2
  >>> recognize_numberbj("Wrong card")
  10
  """
  identify = card[:4]

  #If the card is a number
  if identify[0].isnumeric():
    if identify[1].isnumeric():
      return 10
    return int(identify[0])

  #If card is an ace
  if identify[0:3] == "Ace":
    return 11

  #If the card is a face card
  return 10

def draw(deck) -> str:
  """Returns the first index of deck  that has a value while removing said value

  Precondtion: deck is a string list

  >>>draw(["1","2","3"])
  "1"
  >>>draw([null,"s"])
  "s"
  >>>draw([null,null])
  "null"
  """
  holder = ""

  for i in range(len(deck)):
    if deck[i] != "null":
      holder, deck[i] = deck[i], "null"
      return holder
  return "null"  

roulette.py
from termcolor import *

#Table
from termcolor import *

def rowcolor(nums, start):
  if start == 'red':
    nums_colors = ['red', 'grey', 'red']
  else:
    nums_colors = ['grey', 'red', 'grey']
  
  for i in range(2):
    if nums_colors[i] == 'grey':
      print(colored(nums[i], nums_colors[i], 'on_white'), end = " ")
    else:
      print (colored(nums[i], nums_colors[i]), end = " ")
    print ("| ", end = "")

  if nums_colors[-1] == "red":
    print(colored(nums[-1], nums_colors[-1]))
  else:
    print(colored(nums[-1], nums_colors[-1], "on_white"))

def rowcolor_free(nums, color):
  for i in range(2):
    if color[i] == "grey":
      print(colored(nums[i], color[i], "on_white"), end = "")
    else:
      print(colored(nums[i], color[i]), end = "")
    print("| ", end = "")
  
  if color[-1] == "red":
    print(colored(nums[-1], color[-1]))
  else:
    print(colored(nums[-1], color[-1], "on_white"))

def add_row(row):
  print("Row " + str(row) + ": " , end = "")
  return row +1

def table():
  print(colored("          0", 'green'), end =" ")
  print("| ", end = "")
  print(colored("00", 'green'))

  row = 1
  r = "red"
  g = "grey"

  for i in range(1, 9, 3):
    row = add_row(row)
    print(" ", end = "")

    if i%2 == 1:
      rowcolor([i, i+1, i+2], "red")
    else:
      rowcolor([i, i+1, i+2], "grey")

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([10, 11, 12], [g, g, r])

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([13, 14, 15], [g, r, g])

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([16, 17, 18], [r, g, r])

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([19, 20, 21], [r, g, r])

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([22, 23, 24], [g, r, g])

  row = add_row(row)
  print(" ", end = "")
  rowcolor_free([25, 26, 27], [r, g, r])

  row = add_row(row)
  rowcolor_free([28, 29, 30], [g, g, r])

  row = add_row(row)
  rowcolor_free([31, 32, 33], [g, r, g])

  row = add_row(row)
  rowcolor_free([34, 35, 36], [r, g, r])

  print("        C1| C2| C3")


#Choices
def valid(lst, decision, num):
  lst[0] = decision
  lst[1] = int(num)
  lst[2] = 1

def choices():
  decision = [0,0,0]
  print("Singles: 1")
  print("Split: 2")
  print("Street: 3")
  print("Square: 4")
  print("Six: 5")
  print("Colour: 6")
  print("Dozen: 7")
  print("Halfs: 8")
  print("Even/Odd: 9")
  print("Columns: 10")

  pick = input("Enter an option from the above: ")
  if not pick.isnumeric() or not 1 <= int(pick) <= 10:
    print("Enter valid input")
      
  elif pick == "1":
    num = input("Enter a number between 0-36 or 37 for 00: ")
    if not num.isnumeric() or not 0 <= int(num) <= 37:
      print("Enter valid input")

    else:
      valid(decision, "Single", num)
  
  elif pick == "2":
    num = input("Enter the first number: ")
    if not num.isnumeric() or not 1 <= int(num) <= 36:
      print("Enter valid input")

    else:
      ver = input("Enter 1 if it is a split to the right or 2 if the split is downwards: ")
      if not ver.isnumeric() or not 1 <= int(ver) <= 2:
        print("Enter valid input")      

      else:
        if int(ver) == 1:
          if int(num) % 3 == 0:
            print("Enter valid input")

          else:
            valid(decision, "Split1", num)
        
        else:
          if 34 <= int(num) <= 36:
            print("Enter valid input")

          else:
            valid(decision, "Split2", num)
  
  elif pick == "3":
    num = input("Enter the row you want to bet on: ")
    if not num.isnumeric() or not 1 <= int(num) <= 12:
      print("Enter valid input")
    else:
      valid(decision, "Street", num)
  
  elif pick == "4":
    num = input("Enter the top left of the square: ")
    if not num.isnumeric() or not int(num)%3 == 0 or not 34 <= int(num) <= 36:
      print("Enter valid input")
    else:
      valid(decision, "Square", num)
  
  elif pick == "5":
    num = input("Enter the top left of the six numbers: ")
    if not num.isnumeric() or not (int(num)-1)%3 == 0 or int(num) == 34:
      print("Enter valid input")

    else:
      valid(decision, "Six", num)
    
  elif pick == "6":
    num = input("Enter 'Black' or 'Red': ")
    if num == "Black":
      valid(decision, "Black", 0)

    else:
      valid(decision, "Red", 0)
    
  elif pick == "7":
    num = input("Enter 1, 2, or 3 to pick column 1, 2, or 3 respectivley: ")

    if num.isnumeric() or not 1 <= int(num) <= 3:
      print("Enter valid input")

    else:
      valid(decision,"Dozen " + num, 0)
    
  elif pick == "8":
    num = input("Enter 1 for 1 - 18 or 2 for 19 - 36: ")

    if not num.isnumeric() or not 1 <= int(num) <= 2:
      print("Enter valid input")
      
    elif num == "1":
      valid(decision, "1 - 18", 0)

    else:
      valid(decision, "19 - 36", 0)
    
  elif pick == "9":
    num = input("Enter 1 for even or 2 for odd: ")
    if not num.isnumeric() or not 1 <= int(num) <= 2:
      print("Enter valid input")
      
    elif num == "1":
      valid(decision, "Even", 0)

    else:
      valid(decision, "Odd", 0)
    
  elif pick == "10":
    num = input("Enter 1, 2, or 3 for the coresponding column: ")
    if not num.isnumeric() or not 1 <= int(num) <= 3:
      print("Enter valid input")
      
    else:
      valid(decision, "Column " + num, 0)

  return decision
    



card_game.py
import cards 
import random

def black_jack(bet: int) -> int:
  """
  Simulates a game of blackjack
  """
  #Creates and shuffle deck
  deck = cards.deck_of_cards()
  random.shuffle(deck)    

  #Variables for the game
  users_hand = []
  has_ace = 0 
  users_total = 0  
  
  dealers_hand = []
  dealers_ace = 0
  dealers_total = 0

  #Draw cards
  users_hand.append(cards.draw(deck))
  users_hand.append(cards.draw(deck))

  dealers_hand.append(cards.draw(deck))
  dealers_hand.append(cards.draw(deck))
  
  #Identifies Aces
  for i in users_hand:
    if i[0:3] == "Ace":
      has_ace += 1
  for i in dealers_hand:
    if i[0:3] == "Ace":
      dealers_ace += 1

  #Create help to identify and add value of card
  users_total = cards.recognize_numberbj(users_hand[0]) + cards.recognize_numberbj(users_hand[1])

  dealers_total = cards.recognize_numberbj(dealers_hand[0]) + cards.recognize_numberbj(dealers_hand[1])

  #If hand has two aces
  if users_total > 21:
    users_total = users_total - 10
    has_ace = has_ace - 1
  if dealers_ace > 21:
    dealers_total = dealers_total - 10
    dealers_ace = dealers_ace - 1

  #Show draws
  print("User's hand: ", end = "")
  print(users_hand)
  print("User's total: " + str(users_total))
  print()

  print("Dealer's hand: ", end = "")
  print("['" + dealers_hand[0] + "', 'Hidden Card']")
  print("Dealer's total: " + str(cards.recognize_numberbj(dealers_hand[0])))
  print("========================================")
  
  #Outcome of deal
  #If you have natural
  if users_total == 21:
    print("You have a natural so it is time to reveal the dealer's hand")
    print("Dealer's hand: ", end = "")
    print(dealers_hand)
    print("Dealer's total: " +  str(dealers_total))

    if dealers_total == 21:
      print("It is a tie!")
      return 0
    print("You Win!")
    return bet

  #Otherwise
  else:
    #Player's turn
    while input("Hit or Stay: ") != "Stay":
      #Draw Card
      users_hand.append(cards.draw(deck))
      users_total += cards.recognize_numberbj(users_hand[-1])

      if cards.recognize_numberbj(users_hand[-1]) == 11:
        has_ace += 1

      #Check if user has ace if total over 21
      if users_total > 21 and has_ace != 0:
        users_total = users_total - 10
        has_ace = has_ace - 1

      #Prints out result
      print("User's hand: ", end = "")
      print(users_hand)
      print("User's total: " + str(users_total))
      print()

      #User busted
      if users_total > 21:
        print("You are busted")
        return bet * -1

    #Dealer's turn
    print("========================================")
    print ("You stayed so now it's time for the dealer")
    print("Dealer's hand: ", end = "")
    print(dealers_hand)
    print("Dealer's total: " +  str(dealers_total))
    print()
    
    while dealers_total < 17:
      #Draw card
      dealers_hand.append(cards.draw(deck))
      dealers_total += cards.recognize_numberbj(dealers_hand[-1])

      if cards.recognize_numberbj(dealers_hand[-1]) == 11:
        dealers_ace += 1

      #Check if dealer has ace given total is over 21
      if dealers_ace != 0 and dealers_total > 21:
        dealers_total = dealers_total - 10
        dealers_ace = dealers_ace - 1

      #Reveal outcome
      print("Dealer's hand: ", end = "")
      print(dealers_hand)
      print("Dealer's total: " +  str(dealers_total))
      print()

      #If dealer busted
      if dealers_total > 21:
        print("Dealer has busted")
        return bet

    print("========================================")

    #If dealer and user stayed
    if dealers_total == users_total:
      print("Both hands are equal. It's a tie.")
      return 0

    if dealers_total > users_total:
      print("Dealer has the higher hand. You lose.")
      return bet * -1

    print("You have the higher hand. You win!")
    return bet


other_games.py
import random

def roulette(bet: str, money: int, number = None) -> int:
  """
  Returns if the user's bet is correct in a simulation of roulette
  """
  #Outcomes
  red = [1,3,5,7,9,12,14,16,18,19,21,23,25,27,30,32,34,36]
  black = [2,4,6,8,10,11,13,15,17,20,22,24,26,28,29,31,33,35]
  column_one = [1,4,7,10,13,16,19,22,25,28,31,34]
  column_two = [2,5,8,11,14,17,20,23,26,29,32,35]
  column_three = [3,6,9,12,15,18,21,24,27,30,33,36] 
  users = []

  #Roll and print outcome
  outcome = random.randint(0,37)  
  
  if outcome == 37:
    print("The number is 00")
  else:
    print("The number is " + str(outcome))

  #Singles 35 - 1 
  if bet == "Single":
    if outcome == int(number):
      return money * 35

    else:
      return -1 * money

  #Splits 17 - 1
  if "Split" in bet:
    if "1" in bet:
      users = [number, number + 1]

    else:
      users = [number, number + 3]

    if outcome in users:
      return money * 17
  
  #Street 11 - 1
  if "Street" in bet:
    users = [number*3, number*3-1, number*3-2]
    if outcome in users:
      return money * 11

  #Square 8 - 1
  if "Square" == bet:
    users = [number, number+1, number+3, number+4]
    if outcome in users:
      return money * 8

  #Six 5 - 1
  if "Six" == bet:
    for i in range(6):
      users.append(number + i)

    if outcome in users:
      return money * 5

  #Colours 1 - 1
  elif bet == "Red":
    if outcome in red:
      return money

  elif bet == "Black":
    if outcome in black:
      return money
  
  #Dozens 2 - 1
  elif bet == "Dozen 1":
    if 1 < outcome <= 12:
      return money * 2
  
  elif bet == "Dozen 2":
    if 13 <= outcome <= 24:
      return money * 2

  elif bet == "Dozen 3":
    if 25 <= outcome < 37:
      return money * 2
  
  #Highlow 1 - 1
  elif bet == "1 - 18":
    if outcome == 0 or outcome == 37 or outcome >= 19:
      return money * -1

    return money
  
  elif bet == "19 - 36":
    if outcome == 0 or outcome == 37 or outcome <= 18:
      return money * -1

    return money

  #Even/Odd 1 - 1
  elif bet == "Even":
    if outcome == 0 or outcome == 37 or outcome % 2 == 1:
      return money * -1

    return money

  elif bet == "Odd":
    if outcome == 0 or outcome == 37 or outcome % 2 == 0:
      return money * -1

    return money 
  
  #Column 2 - 1
  elif bet == "Column 1":
    if outcome in column_one:
      return money * 2

  elif bet == "Column 2":
    if outcome in column_two:
      return money * 2

  elif bet == "Column 3":
    if outcome in column_three:
      return money * 2
  
  return money * -1

def high_low(guess: str, bet: int) -> int:
  """
  Returns if guess is correct when comparing num and roll

  """

  #Creates and print outcome
  roll = random.randint(1,100)
  print ("You rolled a " + str(roll))

  #Check if user's guess is correct
  if 50 < roll and guess == "high":    
    return bet

  elif 50 > roll and guess == "low":
    return bet

  return -1 * bet

def craps(shooter_outcome: str, bet: int) -> int:
  """
  Returns if the player guesses correctly in a simulation of the game Craps
  """
  #Variables for the game
  die1 = random.randint(1,6)
  die2 = random.randint(1,6)
  total = die1 + die2
  breakout = True

  #Print the outcome
  print("your rolls are: " + str(die1) + "," + str(die2))
  print("The total is " + str(total))
  print()
  
  #Comeout roll results
  #If the roll is a natural
  if total == 7 or total == 11:
    print("A natural!")
    if shooter_outcome == "Win":
      print("You guessed right!")
      return bet 
    print("You guessed wrong.")
    return bet * -1

  #If the roll is a crap
  elif total <= 3:
    print("It's a crap!")
    if shooter_outcome == "Win":
      print("You guessed wrong.")
      return bet * -1
    print("You guessed right!")
    return bet

  #If the roll is a push 
  elif total == 12:
    return 0

  #If the roll was something other than the rolls above
  point = total
  print("Your point is " + str(point))
  print()

  #Played until the outcome is either the number they rolled in the come out or is 7
  while breakout:
    die1 = random.randint(1,6)
    die2 = random.randint(1,6)
    total = die1 + die2

    print("your rolls are: " + str(die1) + "," + str(die2))
    print("The total is " + str(total))
    print()

    if total == 7:
      if shooter_outcome == "Win":
        print("You guessed wrong.")
        return bet * -1
      print("You guessed right!")
      return bet

    elif total == point:
      if shooter_outcome == "Win":
        print("You guessed right!")
        return bet
      print("You guessed wrong.")
      return bet * -1